### ISSUE TITLE
NumberControl: Increment and Decrement buttons aren't operable with keyboard

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#57401]
### Description

I'm not sure what is the reasoning behind making the NumberControl Increment and Decrement buttons not operable with a keyboard. That seems intentional as:
- These buttons have a `tabIndex="-1"` attribute so they're not focusable.
- They are hidden from assistive technology by the means of `aria-hidden="true"`.
- However, they do have an `aria-label` attribute.

Regardless of the inconsistent implementation, there is no apparent reason why these buttons should not be usable with a keyboard.

Screenshot of an instance of NumberControl Increment and Decrement buttons:

![Screenshot 2023-12-27 at 10 32 41](https://github.com/WordPress/gutenberg/assets/1682452/504ca3bc-e3b8-4fb5-9311-90a38fdfce09)

### Step-by-step reproduction instructions

- Use Twenty Twenty Four as active theme
- Go to Site Editor > Templates > Blog Home
- Use the Tab key to move focus to the 'Posts per page' field.
- Press the Tab key again.
- Observe the Increment and Decrement buttons get skipped bevause they're made not focusable.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
For reference: the custom spin controls were adder here, with a comment mentioning them as progressive enhancement and hidden from screen readers.

https://github.com/WordPress/gutenberg/pull/45333
> The buttons are a progressive enhancement hidden from screen readers.

The progressive enhancement concept is totally fine but there's no reason to make them not operable with a keyboard.
